lib/rspec/support.rb

```diff
@@ -83,11 +83,11 @@ def self.failure_notifier=(callable)
     end
 
     def self.failure_notifier
-      thread_local_data[:failure_notifier] ||= method(:raise)
+      thread_local_data[:failure_notifier] ||= lambda { |failure, _opts| method(:raise)(failure) }
```

Comment:
This isn't valid syntax...

And actually, there's no reason to capture the `raise` method only to call it right away -- we can just send the `raise` method directly.  This could be:

``` ruby

    DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }

    def self.failure_notifier
      thread_local_data[:failure_notifier] ||= DEFAULT_FAILURE_NOTIFIER
    end
```